app/sign.js

```diff
@@ -276,13 +277,45 @@ const handleSignErc20 = function(recoveryRequest, key, skipConfirm) {
   return transaction.serialize().toString('hex');
 };
 
+/* *
+  Takes in either an xprv, xlmsecret, or 24 words.
+  Returns an xprv or xlmsecret
+ */
+const parseKey = function(rawkey, coin, path) {
+
+  if(rawkey.includes(',') && rawkey.split(',').length === 24) {
+    const mnemonic = rawkey.replace(/,/g,' '); // replace commas with spaces
+    if(coin === 'xlm' || coin === 'txlm') {
+      // stellar is special (thanks Jeb)
+      const stellarWallet = stellarHd.fromMnemonic(mnemonic);
+      return stellarWallet.getSecret(0);
+    }
+
+    // every other coin can use xpubs
+    if(!bip39.validateMnemonic(mnemonic)) {
+      throw new Error("Invalid mnemonic");
+    }
+    const seed = bip39.mnemonicToSeed(mnemonic);
+    let node = utxoLib.HDNode.fromSeedBuffer(seed);
+    if(path) {
+      node = node.derivePath(path);
+    }
+    const xprv = node.toBase58();
+    return xprv;
+
+  }
+  // if it doesn't have commas, we expect it is an xprv or xlmsecret properly formatted
```

Comment:
or it is a mnemonic separated by spaces (which I believe is standard). Is that going to cause an issue?

Reply:
Correct the mnemonic is separated by spaces when getting an xprv. I just have the user input the words separated by commas because then they can be passed in as one argument on the command line. First thing that is done in this parseKey code is replace the commas with spaces.
example usage:
./admin.js sign --key dog,cat,rabbit,piano,word,word,word